File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/java_solutions/2087-minimum-cost-homecoming-of-a-robot-in-a-grid.java
class Solution {
    public int minCost(int[] startPos, int[] homePos, int[] rowCosts, int[] colCosts) {
        long totalCost = 0;

        int startRow = startPos[0];
        int startCol = startPos[1];
        int homeRow = homePos[0];
        int homeCol = homePos[1];

        // Calculate cost for row movements
        if (startRow < homeRow) {
            for (int r = startRow + 1; r <= homeRow; r++) {
                totalCost += rowCosts[r];
            }
        } else if (startRow > homeRow) {
            for (int r = startRow - 1; r >= homeRow; r--) {
                totalCost += rowCosts[r];
            }
        }

        // Calculate cost for column movements
        if (startCol < homeCol) {
            for (int c = startCol + 1; c <= homeCol; c++) {
                totalCost += colCosts[c];
            }
        } else if (startCol > homeCol) {
            for (int c = startCol - 1; c >= homeCol; c--) {
                totalCost += colCosts[c];
            }
        }

        return (int) totalCost;
    }
}

File: leetcode/AIcoding/3.DeepSeek-v3/go_solutions/2087-minimum-cost-homecoming-of-a-robot-in-a-grid.go
func minCost(startPos []int, homePos []int, rowCosts []int, colCosts []int) int {
    cost := 0
    x1, y1 := startPos[0], startPos[1]
    x2, y2 := homePos[0], homePos[1]

    if x1 < x2 {
        for i := x1 + 1; i <= x2; i++ {
            cost += rowCosts[i]
        }
    } else if x1 > x2 {
        for i := x1 - 1; i >= x2; i-- {
            cost += rowCosts[i]
        }
    }

    if y1 < y2 {
        for j := y1 + 1; j <= y2; j++ {
            cost += colCosts[j]
        }
    } else if y1 > y2 {
        for j := y1 - 1; j >= y2; j-- {
            cost += colCosts[j]
        }
    }

    return cost
}